netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -185,6 +189,23 @@ private void onRstStreamRead(int streamId)
     stream.transportReportStatus(Status.UNKNOWN, false, new Metadata.Trailers());
   }
 
+  private void onGoAwayRead(long errorCode, ByteBuf debugData) {
+    Status status = HttpUtil.Http2Error.statusForCode((int) errorCode);
+    if (debugData.isReadable()) {
+      // If a debug message was provided, use it.
+      String msg = debugData.toString(UTF_8);
+      status = status.withDescription(msg);
```

Comment:
It seems this should have been augmentDescription as well, so that we know what the error code was.